text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+
+An Entity Detector would have an API similar to:
+
+```rust
+trait EntityDetector
+  pub fn detect_entities(...) -> Result<Vec<Entity>, EntityDetectionError>>
+```
+
+Where `Result` is the equivalent of error channel in the language of choice (e.g. in Go this would be `entities, err := e.detectEntities()`).
+
+#### Entity Merging and Resource
+
+The most important aspect of this design is how Entities will be merged to construct a Resource. We provide a simple algorithm for this behavior:
+
+- Construct a set of detected entities, E
+- All entity detectors are sorted by priority
+- For each entity detector
+  - For each entity detected
+    - If the entity exists in E, ignore it
+    - Otherwise, add the entity to E
+- Construct a Resource from the set E.
+
+Any implementation that achieves the same result as this algorithm is acceptable.
+
+#### Environment Variable Detector
+
+An Entity detector will be specified to allow Platform to inject entity identity information into workloads running on that platform.   For Example, the OpenTelemetry Operator could inject information about Kubernetes Deployment + Container into the environment, which SDKs can elect to interact with (through configuration of the Environment Variable Entity Detector).
+
+While details of ENV variables will be subject to change, it would look something like the following:
+
+```bash
+set OTEL_DETECTED_ENTITIES=k8s.deployment[k8s.deployment.name=my-program],k8s.pod[k8s.pod.name=my-program-2314,k8s.namespace=default]
+<run my program>
+```
+
+The minimum requirements of this entity detector are:
+
+- ENV variable can specify multiple entities (resource attribute bundles)
+- ENV variable can be easily appended or leverages by multiple participating systems, if needed.
+- Entities discovered via ENV variable can participate in Resource Manager generically, i.e. resolving conflicting definitions.
+
+The actual design for this ENV variable interaction would follow the approval of this OTEP.
+
+### Interactions with OpenTelemetry Collector
+
+The OpenTelemetry collector can be updated to optionally  interact with Entity on Resource. A new entity-focused resource detection process can be created which allows add/override behavior at the entity level, rather than individual attribute level.
+
+For example, the existing resource detector looks like this:
+
+```yaml
+processors:
+  resourcedetection/docker:
+    detectors: [env, docker]
+    timeout: 2s
+    override: false
+```
+
+The future entity-based detector would look almost exactly the same, but interact with the entity model of resource:
+
+```yaml
+processor:
+  entityresourcedetection:
```

Comment:
I think instead of introducing a new `entityresourcedetection` processor we should aim to modify `resourcedetection` processor in backward compatible manner (preserving it's old behavior of Resource detection) and add entity detection capabilities. See how it's done in this prototype: https://github.com/tigrannajaryan/opentelemetry-collector-contrib/pull/1/files